agent/taskresource/credentialspec/credentialspec_linux.go

```diff
@@ -0,0 +1,438 @@
+		credentialSpecContainerMap: credentialSpecContainerMap,
+		ServiceAccountInfoMap:      make(map[string]ServiceAccountInfo),
+	}
+
+	s.initStatusToTransition()
+	return s, nil
+}
+
+// Create is used to retrieve credentialspec resources for a given task
+func (cs *CredentialSpecResource) Create() error {
+	var iamCredentials credentials.IAMRoleCredentials
+
+	executionCredentials, ok := cs.credentialsManager.GetTaskCredentials(cs.getExecutionCredentialsID())
+	if ok {
+		iamCredentials = executionCredentials.GetIAMRoleCredentials()
+	}
+
+	var wg sync.WaitGroup
+	errorEvents := make(chan error, len(cs.credentialSpecContainerMap))
+	for credSpecStr := range cs.credentialSpecContainerMap {
+		credSpecSplit := strings.SplitAfterN(credSpecStr, "credentialspec:", 2)
+		if len(credSpecSplit) != 2 {
+			seelog.Errorf("Invalid credentialspec: %s", credSpecStr)
+			continue
+		}
+
+		credSpecValue := credSpecSplit[1]
+		if strings.HasPrefix(credSpecValue, "file://") {
+			wg.Add(1)
+			go cs.handleCredentialspecFile(credSpecStr, &wg, errorEvents)
+			continue
+		}
+
+		parsedARN, err := arn.Parse(credSpecValue)
+		if err != nil {
+			cs.setTerminalReason(err.Error())
+			return err
+		}
+		parsedARNService := parsedARN.Service
+		switch parsedARNService {
+		case "s3":
+			wg.Add(1)
+			go cs.handleS3CredentialspecFile(credSpecStr, credSpecValue, iamCredentials, &wg, errorEvents)
+		case "ssm":
+			wg.Add(1)
+			go cs.handleSSMCredentialspecFile(credSpecStr, credSpecValue, iamCredentials, &wg, errorEvents)
+		default:
+			err = errors.New("unsupported credentialspec ARN, only s3/ssm ARNs are valid")
+			cs.setTerminalReason(err.Error())
+			return err
+		}
+	}
+	wg.Wait()
+	close(errorEvents)
+	if len(errorEvents) > 0 {
+		var terminalReasons []string
+		for err := range errorEvents {
+			terminalReasons = append(terminalReasons, err.Error())
+		}
+
+		errorString := strings.Join(terminalReasons, ";")
+		cs.setTerminalReason(errorString)
+		return errors.New(errorString)
+	}
+
+	seelog.Infof("credentials fetcher daemon request: %v", cs.credentialsFetcherRequest)
+
+	// Check if skip credential fetcher invocation check override is present
+	skipSkipCredentialsFetcherInvocationCheck := utils.ParseBool(os.Getenv(envSkipCredentialsFetcherInvocation), false)
+	if skipSkipCredentialsFetcherInvocationCheck {
+		seelog.Info("Skipping credential fetcher invocation based on environment override")
+		testKrbFilePath := "/tmp/tgt"
+		os.Create(testKrbFilePath)
+		// assign temporary variable for test
+		cs.leaseid = "12345"
+		for k := range cs.ServiceAccountInfoMap {
+			cs.CredSpecMap[k] = testKrbFilePath
+		}
+		return nil
+	}
+
+	err := cs.handleKerberosTicketCreation()
+	if err != nil {
+		cs.setTerminalReason(err.Error())
+		return err
+	}
+
+	return nil
+}
+
+func (cs *CredentialSpecResource) handleKerberosTicketCreation() error {
+	// Create kerberos tickets for the gMSA service accounts on the host location /var/credentials-fetcher/krbdir
```

Comment:
please remind me again - is this directory configurable?

Reply:
as of now it is not configurable